lib/pure/strutils.nim

```diff
@@ -2860,6 +2860,44 @@ proc isNilOrWhitespace*(s: string): bool {.noSideEffect, procvar, rtl,
   ## Alias for isEmptyOrWhitespace
   result = isEmptyOrWhitespace(s)
 
+
+since (1, 1):
+  func toLowerAsciiInPlace*(s: var string) {.inline.} =
+    ## Converts string `s` into lower case.
+    ##
+    ## This works only for the letters ``A-Z``. See `unicode.toLower
+    ## <unicode.html#toLower,string>`_ for a version that works for any Unicode
+    ## character.
+    ##
+    ## See also:
+    ## * `normalize proc<#normalize,string>`_
+    runnableExamples:
+      var x = "abcdefghijklmnopqrstuvwxyz01234567890ABCDEFGHIJKLMNOPQRSTUVWXYZ*+,-./:;<=>?@[]_{|}~"
+      toLowerAsciiInPlace(x)
+      doAssert x == "abcdefghijklmnopqrstuvwxyz01234567890abcdefghijklmnopqrstuvwxyz*+,-./:;<=>?@[]_{|}~"
+    const t = "abcdefghijklmnopqrstuvwxyz".indent(65).cstring
```

Comment:
Silly, pointless lookup table. Instead use ``c = chr(ord(c) - ord('A') + ord('a'))``. Basic math, it's a thing.

Reply:
is slower on my bench, but if you say so :man_shrugging:

Reply:
Microbenchmarks are misleading, in reality you can also have I-cache problems and then you appreciate not everything was "optimized" by 8 times loop unrolling with SSE instructions and 1KB lookup tables.